### ISSUE TITLE
Android credential manager

### ISSUE THREAD

**Issue by jnm2** [jnm2/GitNotified#3]
Branch [test-account-auth](https://github.com/jnm2/GitNotified/tree/test-account-auth)

**Comment by belac9615** [comment]
@jnm2 Is this just the android credential manager implementation, or that and the Oauth service? Can't we make Oauth a part of the shared code? If so, I'll go ahead and build out Oauth.

**Comment by jnm2** [comment]
@belac9615 No, this is just for Android settings UI. It'll have to plug into the OAth that you're building, but I need the Android account with a dummy implementation to be able to explore using the Android sync adapter technology.

**Comment by belac9615** [comment]
@jnm2 Is this complete? I think we finished this when we merged with dev last night.

**Comment by jnm2** [comment]
Yes, though a potential open question is how the android app enumerates all GitNotified accounts to access the tokens.

**Comment by jnm2** [comment]
To avoid needing the `GET_ACCOUNTS` permission and all the logic in [AccountUtils.cs](https://github.com/GitNotified/GitNotified/commit/b9fabcf414a4b3c17b6bbc72f52a27d826353864#diff-f3a5ff906675faa2adafd586165da38b) which tries to manage the display name vs unique ID impedance mismatch, we will have to store the unique ID and account names together in our app's storage.

It would be simplest to skip the Android accounts page integration altogether. We will have a consistent experience 100% in-app for managing logins which won't be mixed in with accounts from other types of apps. https://developer.android.com/training/articles/security-tips.html#Credentials says "Alternatively, if only one application uses the credential, you might use a KeyStore for storage."

So I'm in favor of dropping the Android account integration. We chose not to go with the sync method which would require this integration anyway.

For security purposes we should probably keep the tokens in KeyStore then and everything else in our main storage.

**Comment by jnm2** [comment]
TODO:

- [ ] Remove the AccountAuthenticator and replace it with KeyStore and local storage

**Comment by jnm2** [comment]
KeyStore can't be used to store arbitrary strings. The arbitrary strings will have to be encrypted and decrypted by a key generated and held by the KeyStore, but stored somewhere else. This makes sense.
